enterprise/server/backends/userdb/userdb.go

```diff
@@ -1162,3 +1190,11 @@ func hasAdminOnlyCapabilities(capabilities []akpb.ApiKey_Capability) bool {
 	}
 	return false
 }
+
+func getEmailDomain(email string) string {
+	chunks := strings.Split(email, "@")
+	if len(chunks) == 0 {
+		return ""
+	}
+	return chunks[len(chunks)-1]
```

Comment:
No strong preference here, but another way to do this if you expect two parts is to use `strings.SplitN`

```
parts := strings.SplitN(email, "@", 2)
if len(parts) != 2 {
  return ""
}
return parts[1]
```